app/src/main/java/org/dhis2/data/forms/dataentry/fields/option_set/OptionSetViewModel.java

```diff
@@ -0,0 +1,205 @@
+package org.dhis2.data.forms.dataentry.fields.option_set;
+
+import androidx.annotation.NonNull;
+
+import com.google.auto.value.AutoValue;
+
+import org.dhis2.data.forms.dataentry.fields.FieldViewModel;
+import org.hisp.dhis.android.core.common.ObjectStyle;
+import org.hisp.dhis.android.core.common.ValueTypeDeviceRendering;
+import org.hisp.dhis.android.core.option.Option;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@AutoValue
+public abstract class OptionSetViewModel extends FieldViewModel {
+
```

Comment:
can we use a kotlin dataclass?

Reply:
Nope. Kotlin data classes do not support inheritance